src/Domain/Uuid/Uuid.ts

```diff
@@ -0,0 +1 @@
+export type Uuid = string
```

Comment:
no need to alias primitives - introduces complexity

Reply:
It does, but an absolute minimum amount of it (esp. compared to introducing a class) and it has the following advantages:

* Using `Uuid` instead of `string` documents intention, introduces clarity; easier to spot a mistake
* Changing the type to a more complex (or different) one in the future will not require adjusting annotations
* Some languages and type systems, e.g. ([Flow](https://flow.org/en/docs/types/opaque-types/), [Scala](https://dotty.epfl.ch/docs/reference/other-new-features/opaques.html) introduce opaque types and similar mechanisms to have even stronger abstractions for ensuring type-safety in use cases like this one

this is my generic response to a generic statement ;)

that said, if you feel that in this case the abstraction is not worth it, I can remove it, but maybe there is no point in bothering

In general I might have overabstracted here and there indeed, but doing my best to work within the constraints.

Reply:
We do the same in SNJS fwiw: https://github.com/standardnotes/snjs/blob/master/packages/snjs/lib/types.ts#L2

Reply:
Valid arguments, let's leave it as it is in that case